### PR TITLE
[Refactor:SubminiPolls] Remove status from database poll table

### PR DESCRIPTION
**Pull request by zacharymnp**
### Please check if the PR fulfills these requirements:

* [ ] Tests for the changes have been added/updated (if possible)
* [ ] Documentation has been updated/added if relevant
* [ ] Screenshots are attached to Github PR if visual/UI changes were made

### What is the current behavior?

The 'status' column in the database 'poll' table was deprecated in PR #10184 (Poll-Timer Backend), but it was not removed.

### What is the new behavior?
Removed the 'status' column with no change in functionality.

### CODE REVIEW

File: migration/migrator/migrations/course/20240626171518_remove_poll_status.py [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+    :param config: Object holding configuration details about Submitty
+    :type config: migrator.config.Config
+    :param database: Object for interacting with given database for environment
+    :type database: migrator.db.Database
+    :param semester: Semester of the course being migrated
+    :type semester: str
+    :param course: Code of course being migrated
+    :type course: str
+    """
+    database.execute("ALTER TABLE ONLY polls ADD COLUMN IF NOT EXISTS status TEXT NOT NULL")
```
Comment (williamjallen) [inline migration/migrator/migrations/course/20240626171518_remove_poll_status.py:33]: The column was nullable before, so shouldn't you drop the not null constraint here?

**Review by williamjallen — CHANGES_REQUESTED** [review]
One minor tweak and then this should be good to go.

File: migration/migrator/migrations/course/20240626171518_remove_poll_status.py (hunk 1 above)
Comment (zacharymnp) [reply migration/migrator/migrations/course/20240626171518_remove_poll_status.py]: changed in [caf8eac](https://github.com/Submitty/Submitty/pull/10626/commits/caf8eac1dc7294871046c851350f15781819e2c1)

File: migration/migrator/migrations/course/20240626171518_remove_poll_status.py [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+    :param config: Object holding configuration details about Submitty
+    :type config: migrator.config.Config
+    :param database: Object for interacting with given database for environment
+    :type database: migrator.db.Database
+    :param semester: Semester of the course being migrated
+    :type semester: str
+    :param course: Code of course being migrated
+    :type course: str
+    """
+    database.execute("ALTER TABLE ONLY polls DROP COLUMN IF EXISTS status")
```
Comment (williamjallen) [inline migration/migrator/migrations/course/20240626171518_remove_poll_status.py:17]: One more quick question: is `ONLY` needed here for some reason?  (the same comment applies to the down migration as well)

**Review by williamjallen — CHANGES_REQUESTED** [review]

**Review by williamjallen — APPROVED** [review]

**Pull request merged**